packages/site/src/components/ChainConfig.tsx

```diff
@@ -0,0 +1,308 @@
+          },
+        });
+
+        setChainConfigs(configs as ChainConfigs);
+        // eslint-disable-next-line @typescript-eslint/no-shadow
+      } catch (error) {
+        setError(error as Error);
+      }
+    };
+    // eslint-disable-next-line @typescript-eslint/no-shadow
+    getChainConfig().catch((error) => setError(error));
+  }, []);
+
+
+  useEffect(() => {
+    // set default values for unknown chain
+    if (!chainSelected && !chainConfigs[chainSelected as string]) {
+      setChainConfigs({
+        ...chainConfigs,
+        [chainSelected as string]: {
+          simpleAccountFactory: '',
+          entryPoint: '',
+          bundlerUrl: '',
+          customVerifyingPaymasterPK: '',
+          customVerifyingPaymasterAddress: '',
+        },
+      });
+    }
+  }, [chainSelected]);
+
+  const updateSpecificChainConfig = (
+    chainId: string,
+    configKey: keyof ChainConfig,
+    value: string,
+  ) => {
+    setChainConfigs({
+      ...chainConfigs,
+      [chainId]: {
+        ...chainConfigs[chainId],
+        [configKey]: value,
+      },
+    });
+  };
+
+  const updateChainConfig = async () => {
+    setError(undefined);
+    setSuccessMessage(undefined);
+    if (!chainSelected || !chainConfigs[chainSelected]) {
+      return;
+    }
+    try {
+      const request: KeyringRequest = {
+        id: uuid.v4(),
+        scope: '',
+        account: uuid.v4(),
+        request: {
+          method: 'snap.internal.setConfig',
+          params: [chainConfigs[chainSelected] as ChainConfig],
+        },
+      };
+      await client.submitRequest(request);
+      setSuccessMessage('Chain Config Updated');
+      // eslint-disable-next-line @typescript-eslint/no-shadow
+    } catch (error) {
+      setError(error as Error);
+    }
+  };
+
+  return (
+    <ChainConfigContainer>
+      <ChainConfigItem>
+        <ChainConfigHeader>{'Chain Configuration'}</ChainConfigHeader>
+        <ChainConfigContent isOpen>
+          <ChainConfigErrorContainer>
+            {error && <p>{error.message}</p>}
+          </ChainConfigErrorContainer>
+          <ChainConfigSuccessContainer>
+            {successMessage && <p>{successMessage}</p>}
+          </ChainConfigSuccessContainer>
+          <Select
+            value={chainSelected ?? ''}
+            onChange={(event) => {
+              setChainSelected(event.target.value);
+            }}
+          >
+            <SelectItem disabled value="">
+              {'Select Chain'}
+            </SelectItem>
+            {Object.keys(chainConfigs).map((option: string) => (
+              <SelectItem value={option} key={option}>
+                {chainIdToName(option) ?? `Chain ${option}`}
+              </SelectItem>
+            ))}
+          </Select>
+          {chainSelected && (
+            <StyledBox sx={{ flexGrow: 1 }}>
+              <ChainDescription>Simple Account Factory</ChainDescription>
+              <TextField
```

Comment:
What about doing a small loop here like (a bit pseudo-code, you'll get the idea), WDYT?

```typescript
{Object.keys(ChainConfig).map((key){
  return (
    <TextField
      id={key}
      placeholder={
        chainConfigs?.[chainSelected]?[key] ??
        translate(key) ?? key
      }
      value={
        chainConfigs?.[chainSelected]?[key] ?? ''
      }
      onChange={(event) =>
        updateSpecificChainConfig(
          chainSelected,
          key,
          event.target.value
        )
      }
    />
  );
})}
```

Reply:
Thats a good suggestion 👍